tests/Unit/SDK/Resource/Detectors/ServiceTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Unit\SDK\Resource\Detectors;
+
+use AssertWell\PHPUnitGlobalState\EnvironmentVariables;
+use OpenTelemetry\SDK\Resource\Detectors;
+use OpenTelemetry\SemConv\ResourceAttributes;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \OpenTelemetry\SDK\Resource\Detectors\Service
+ */
+class ServiceTest extends TestCase
+{
+    use EnvironmentVariables;
+
+    const UUID_REGEX = '/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
+
+    private Detectors\Service $detector;
+
+    public function setUp(): void
+    {
+        $this->detector = new Detectors\Service();
+    }
+
+    public function tearDown(): void
+    {
+        $this->restoreEnvironmentVariables();
+    }
+
+    public function test_service_get_resource_with_default_service_instance_id(): void
+    {
+        $resource = $this->detector->getResource();
+
+        $this->assertNotEmpty($resource->getAttributes());
+
+        $id = $resource->getAttributes()->get(ResourceAttributes::SERVICE_INSTANCE_ID);
+        $this->assertMatchesRegularExpression(self::UUID_REGEX, $id);
+    }
+
+    public function test_service_get_resource_multiple_calls_same_service_instance_id(): void
```

Comment:
This is multiple calls within the same process, but typically, will most things be running as shared-nothing and be creating a new ID per request?

I think if that is the case, maybe this detector should be opted into rather than available as a default (perhaps more so as it is still marked experimental?)

Reply:
No problem making this opt-in since the PHP process model could cause incorrect values to be generated.

Is there a better approach for sharing an instance ID across the processes, or a more idiomatic way to derive an instance ID in PHP?

Random UUID v4 is the recommended default, but any useful value can be used to generate a UUID v5 per the spec.

Reply:
Yeah, let's make it opt-in for now. re: sharing an id across processes - I think that would need to be worked out per-runtime, if they provide something we can use.
https://www.php.net/php_sapi_name could be useful, and if not, maybe instance id could be _disabled_ for shared-nothing based on sapi?